### PR TITLE
Extended ps

### PR DESCRIPTION
**Pull request by ajayparida**
Support for listen interval and extened power save configuration from shell.

### CODE REVIEW

**Comment by ajayparida** [comment]
@krish2718 @rado17 @sachinthegreen

**Review by rlubos — COMMENTED** [review]
I think we need to start to document those configs more thoroughly. What does "Extended PS" stand for (I'm guessing it's some abbreviaton for power save, but I'd rather know than guess). What's listen interval related to?

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -43,7 +43,8 @@ enum net_request_wifi_cmd {
 	NET_REQUEST_WIFI_CMD_PS_CONFIG,
 	NET_REQUEST_WIFI_CMD_REG_DOMAIN,
 	NET_REQUEST_WIFI_CMD_PS_TIMEOUT,
-	NET_REQUEST_WIFI_CMD_MAX
+	NET_REQUEST_WIFI_CMD_MAX,
+	NET_REQUEST_WIFI_CMD_LISTEN_INTERVAL,
```
Comment (rado17) [inline include/zephyr/net/wifi_mgmt.h:46]: NET_REQUEST_WIFI_CMD_LISTEN_INTERVAL should come after NET_REQUEST_WIFI_CMD_MAX. I see that this has been corrected in the next commit but please correct it in this commit as well

**Review by rado17 — CHANGES_REQUESTED** [review]

File: subsys/net/l2/wifi/wifi_mgmt.c [hunk 2]
```diff
@@ -343,3 +343,37 @@ void wifi_mgmt_raise_twt_sleep_state(struct net_if *iface,
 					iface, INT_TO_POINTER(twt_sleep_state),
 					sizeof(twt_sleep_state));
 }
+
+static int wifi_set_listen_intvl(uint32_t mgmt_request, struct net_if *iface,
```
Comment (chaitu-tk) [inline subsys/net/l2/wifi/wifi_mgmt.c:347]: Please use full interval, I don't think we are saving anything with the short form.

**Comment by ajayparida** [comment]
> I think we need to start to document those configs more thoroughly. What does "Extended PS" stand for (I'm guessing it's some abbreviaton for power save, but I'd rather know than guess). What's listen interval related to?

Thanks, updated info related to configs.

File: include/zephyr/net/wifi_mgmt.h (hunk 1 above)
Comment (rado17) [reply include/zephyr/net/wifi_mgmt.h:46]: > NET_REQUEST_WIFI_CMD_LISTEN_INTERVAL should come after NET_REQUEST_WIFI_CMD_MAX. I see that this has been corrected in the next commit but please correct it in this commit as well

correction:  NET_REQUEST_WIFI_CMD_LISTEN_INTERVAL should come before NET_REQUEST_WIFI_CMD_MAX.

**Review by tkc17 — CHANGES_REQUESTED** [review]
We can combine all power save parameters to either `set_power_save` if applicable before/after association or `set_power_save_mode` if applicable only before association, we can get rid of

* set_power_save_timeout
* Listen interval
* extended PS

Instead keep only below

* set_power_save - before/after
   - on/off
* set_power_save_mode (we could even rename this to set_power_save_preassoc to clarify the meaning)
   - mode
   - listen_interval
   - timeout (need to  document the meaning, currently this is too vague)

**Comment by ajayparida** [comment]
> We can combine all power save parameters to either `set_power_save` if applicable before/after association or `set_power_save_mode` if applicable only before association, we can get rid of
> 
> * set_power_save_timeout
> * Listen interval
> * extended PS
> 
> Instead keep only below
> 
> * set_power_save - before/after
>   
>   * on/off
> * set_power_save_mode (we could even rename this to set_power_save_preassoc to clarify the meaning)
>   
>   * mode
>   * listen_interval
>   * timeout (need to  document the meaning, currently this is too vague)

Will the below be sufficient
    - keep listen_interval & timeout configuration separate (as existing now)
And pass the other command rleated to power save as below
-  set_power_save_mode  legacy/wmm/extended_ps  on/off

As per UMAC team the "timeout" is only preassoc stuff others can be configured at any point of time, though configured value may take effect after connection establishment.

**Comment by ajayparida** [comment]
Changes related to this PR now being handled in https://github.com/zephyrproject-rtos/zephyr/pull/57116

**Pull request closed without merging**